### PR TITLE
LiquidstakeIbc - Autocompounding

### PR DESCRIPTION
**Pull request by kruspy**
## 1. Overview

Module autocompounding

### CODE REVIEW

File: x/liquidstakeibc/keeper/hooks.go [hunk 1]
```diff
@@ -118,26 +123,50 @@ func (k *Keeper) OnRecvIBCTransferPacket(
...
-		ctx,
-		func(u liquidstakeibctypes.Unbonding) bool {
-			return u.UnbondAmount.Denom == hc.HostDenom && u.State == liquidstakeibctypes.Unbonding_UNBONDING_MATURED
-		},
-	)
+	if data.GetSender() == hc.RewardsAccount.Address &&
+		data.GetReceiver() == k.GetDepositModuleAccount(ctx).GetAddress().String() {
+		// add the deposit amount to the deposit record for that chain/epoch
+		currentEpoch := k.GetEpochNumber(ctx, liquidstakeibctypes.DelegationEpoch)
+		deposit, found := k.GetDepositForChainAndEpoch(ctx, hc.ChainId, currentEpoch)
```
Comment (puneet2019) [inline x/liquidstakeibc/keeper/hooks.go:149]: cut autostake fee here?

File: x/liquidstakeibc/keeper/ibc.go [hunk 2]
```diff
@@ -302,10 +303,6 @@ func (k *Keeper) handleSuccessfulAck(
 			if err = k.HandleUndelegateResponse(ctx, msg, msgResponse, channel, sequence); err != nil {
 				return err
 			}
-		case sdk.MsgTypeURL(&ibctransfertypes.MsgTransfer{}):
```
[the remark this answers is not in the document]
Comment (kruspy) [reply x/liquidstakeibc/keeper/ibc.go:305]: nothings needs to be done on ica transfer ack, the code will handle it on the ibc transfer hooks

[the remark this answers is not in the document]
Comment (puneet2019) [reply x/liquidstakeibc/keeper/ibc.go:305]: sequence ID for transfer? again, the PFM can spoof

[the remark this answers is not in the document]
Comment (kruspy) [reply x/liquidstakeibc/keeper/ibc.go:305]: Yeah I will implement that, wasn't checking it before

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**